testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -107,33 +108,34 @@ impl Experiment for ValidatorVersioning {
         // TODO: In the future we may want to pass this functor as an argument to the experiment
         // to make versioning test extensible.
         let txn_payload = TransactionPayload::ScriptFunction(ScriptFunction::new(
-            ModuleId::new(account.address, Identifier::new("PHANTOM_MODULE").unwrap()),
+            ModuleId::new(
+                account.address(),
+                Identifier::new("PHANTOM_MODULE").unwrap(),
+            ),
             Identifier::new("PHANTOM_FUNCTION").unwrap(),
             vec![],
             vec![],
         ));
 
-        let txn_gen = |account: &AccountData| {
-            create_user_txn(
-                &account.key_pair,
-                txn_payload.clone(),
-                account.address,
-                account.sequence_number,
-                123456,
-                0,
-                XUS_NAME.to_owned(),
-                10,
-                ChainId::test(),
+        let txn_gen = |account: &mut LocalAccount| {
+            account.sign_with_transaction_builder(
+                TransactionFactory::new(ChainId::test())
```

Comment:
create the txnfactory outside of this closure and reference it instead of creating a new one each time.